pkg/ddc/base/operation.go

```diff
@@ -149,17 +150,23 @@ func (t *TemplateEngine) reconcileExecuting(ctx cruntime.ReconcileRequestContext
 		log.Error(err, "status handler is nil")
 		return utils.RequeueIfError(err)
 	}
-	opStatus, err = statusHandler.GetOperationStatus(ctx, object, opStatus)
+	opStatusToUpdate, err := statusHandler.GetOperationStatus(ctx, object, opStatus)
 	if err != nil {
 		log.Error(err, "failed to update status")
 		return utils.RequeueIfError(err)
 	}
-	if err = operation.UpdateOperationApiStatus(object, opStatus); err != nil {
-		log.Error(err, "failed to update api status")
-		return utils.RequeueIfError(err)
+	if !reflect.DeepEqual(opStatus, opStatusToUpdate) {
+		if err = operation.UpdateOperationApiStatus(object, opStatusToUpdate); err != nil {
+			log.Error(err, "failed to update api status")
+			return utils.RequeueIfError(err)
+		}
+		if opStatusToUpdate.Phase != common.PhaseExecuting {
```

Comment:
I think UpdateOperationApiStatus will trigger requeue. Why do we need RequeueImmediately?

Reply:
It's a redundant RequeueImmediately and i will remove it.